astropy/io/fits/tests/test_fitstime.py

```diff
@@ -177,6 +164,19 @@ def test_fits_to_time_meta(self, table_types):
         assert tm.meta['MJD-OBS'].value == t.meta['MJD-OBS']
         assert tm.meta['MJD-OBS'].scale == FITS_DEPRECATED_SCALES[t.meta['TIMESYS']]
 
+        # Test for conversion of time data to its value, as defined by its format
+        t['a'].info.serialize_method['fits'] = 'formatted_value'
+        t.write(self.temp('time.fits'), format='fits', overwrite=True)
+        tm = table_types.read(self.temp('time.fits'), format='fits')
+
+        # Test DATE
+        assert not isinstance(tm.meta['DATE'], Time)
+        assert tm.meta['DATE'] == t.meta['DATE']
+
+        # Test MJD-xxx
+        assert not isinstance(tm.meta['MJD-OBS'], Time)
+        assert tm.meta['MJD-OBS'] == t.meta['MJD-OBS']
+
```

Comment:
I think there should be an additional test that `np.all(tn['a'] == t['a'].value)` (since the time has format `isot`, `.value` is a string, and so in comparing there are no worries about rounding errors etc.)